tools/test-js-with-node

```diff
@@ -291,12 +290,66 @@ def run_tests_via_node_js() -> int:
     ret = 0
     if parallel > 1:
         sub_tests = [test_files[i::parallel] for i in range(parallel)]
-        parallel_processes = [subprocess.Popen(node_tests_cmd + sub_test) for sub_test in sub_tests]
+        if options.coverage:
+            # ensure a clean start.
+            coverage_dir = os.path.join(ROOT_DIR, "var/node-coverage")
+            if os.path.exists(coverage_dir) and os.path.isdir(coverage_dir):
+                shutil.rmtree(coverage_dir)
+            # ensure that we work with a unique output folder just in case.
+            random_suffix = random.randrange(2 ** 63)
+            temp_dir = os.path.join(coverage_dir, f"nyc_outputs_{random_suffix}")
```

Comment:
[`tempfile.NamedTemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) does this but better; don’t reinvent it.